standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java

```diff
@@ -55,13 +54,48 @@ public static ExceptionHandler handleException(Exception e) {
   /**
    * Throws if the input exception is the instance of the one in the input classes
    */
-  public <T extends Exception> ExceptionHandler
-      throwIfInstance(Class ...te) throws T {
-    if (te != null) {
-      for (Class<T> t : te) {
-        throwIfInstance(t);
-      }
-    }
+  public <T1 extends Exception,
+          T2 extends Exception>
+  ExceptionHandler throwIfInstance(
+      Class<T1> te1,
+      Class<T2> te2) throws T1, T2 {
+    throwIfInstance(te1);
+    throwIfInstance(te2);
+    return this;
+  }
+
+  /**
+   * Throws if the input exception is the instance of the one in the input classes
+   */
+  public <T1 extends Exception,
+          T2 extends Exception,
+          T3 extends Exception>
+  ExceptionHandler throwIfInstance(
+      Class<T1> te1,
+      Class<T2> te2,
+      Class<T3> te3) throws T1, T2, T3 {
+    throwIfInstance(te1);
+    throwIfInstance(te2);
+    throwIfInstance(te3);
+    return this;
+  }
+
+  /**
+   * Throws if the input exception is the instance of the one in the input classes
+   */
+  public <T1 extends Exception,
+          T2 extends Exception,
+          T3 extends Exception,
+          T4 extends Exception>
+  ExceptionHandler throwIfInstance(
```

Comment:
Nit: maybe we do not need this method with four arguments,  the other three methods seem to be enough for handling exceptions.

Reply:
you are right - when I suspected this issue I changed one of the call sites which already had 3 arguments ...but then undone that change...

removed